plugin/storage/grpc/proto/storage.proto

```diff
@@ -0,0 +1,188 @@
+
+message WriteSpanResponse {
+    oneof response {
+        EmptyResponse success = 1;
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message GetTraceRequest {
+    bytes trace_id = 1 [
+                       (gogoproto.nullable) = false,
+                       (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.TraceID",
+                       (gogoproto.customname) = "TraceID"
+                       ];
+}
+
+message GetTraceSuccess {
+    Trace trace = 1;
+}
+
+message GetTraceResponse {
+    oneof response {
+        GetTraceSuccess success = 1;
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message GetServicesRequest {}
+
+message GetServicesSuccess {
+    repeated string services = 1;
+}
+
+message GetServicesResponse {
+    oneof response {
+        GetServicesSuccess success = 1;
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message GetOperationsRequest {
+    string service = 1;
+}
+
+message GetOperationsSuccess {
+    repeated string operations = 1;
+}
+
+message GetOperationsResponse {
+    oneof response {
+        GetOperationsSuccess success = 1;
+        GrpcStoragePluginError error = 2;
+    }
+}
+
+message TraceQueryParameters {
+    string service_name = 1;
+    string operation_name = 2;
+    map<string, string> tags = 3;
+    google.protobuf.Timestamp start_time_min = 4 [
+                                                 (gogoproto.stdtime) = true,
+                                                 (gogoproto.nullable) = false
+                                                 ];
+    google.protobuf.Timestamp start_time_max = 5 [
+                                                 (gogoproto.stdtime) = true,
+                                                 (gogoproto.nullable) = false
+                                                 ];
+    google.protobuf.Duration duration_min = 6 [
+                                              (gogoproto.stdduration) = true,
+                                              (gogoproto.nullable) = false
+                                              ];
+    google.protobuf.Duration duration_max = 7 [
+                                              (gogoproto.stdduration) = true,
+                                              (gogoproto.nullable) = false
+                                              ];
+    int32 num_traces = 8;
+}
+
+message FindTracesRequest {
+    TraceQueryParameters query = 1;
+}
+
+message FindTraceIDsRequest {
+    TraceQueryParameters query = 1;
+}
+
+message FindTraceIDsSuccess {
+    repeated bytes trace_id = 1 [
+                                (gogoproto.nullable) = false,
+                                (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.TraceID",
+                                (gogoproto.customname) = "TraceIDs"
+                                ];
+}
+
+message FindTraceIDsResponse {
+    oneof response {
+        FindTraceIDsSuccess success = 1;
+        GrpcStoragePluginError error = 2;
```

Comment:
couldn't we instead use [grpc Status](https://jbrandhorst.com/post/grpc-errors/)? It already includes an error message.

Reply:
Something to note here is that the gogo plugin of Status is go 1.11+ (https://github.com/gogo/status)